From d069b7b94287dae15fcaf707775f5dd2bbebcc38 Mon Sep 17 00:00:00 2001
From: Matt McCormick <matt.mccormick@kitware.com>
Date: Wed, 5 Feb 2014 15:52:34 -0500
Subject: [PATCH 1/3] BUG: Wrap TransformFileReader, TransformFileWriter.

Missing CMakeLists.txt in the wrapping directory.  Need to explicitly include
the headers after "Template" was appended to the class name.

Suggested-by: Sami Koho <sami.koho@gmail.com>

Change-Id: I6f994bb18f37d5adf1d0f0311fa73506c41239a7
---
 Modules/IO/TransformBase/wrapping/CMakeLists.txt              | 3 +++
 Modules/IO/TransformBase/wrapping/itkTransformFileReader.wrap | 2 ++
 Modules/IO/TransformBase/wrapping/itkTransformFileWriter.wrap | 2 ++
 3 files changed, 7 insertions(+)
 create mode 100644 Modules/IO/TransformBase/wrapping/CMakeLists.txt

diff --git a/Modules/IO/TransformBase/wrapping/CMakeLists.txt b/Modules/IO/TransformBase/wrapping/CMakeLists.txt
new file mode 100644
index 0000000..6072567
--- /dev/null
+++ b/Modules/IO/TransformBase/wrapping/CMakeLists.txt
@@ -0,0 +1,3 @@
+itk_wrap_module(ITKIOTransformBase)
+itk_auto_load_submodules()
+itk_end_wrap_module()
diff --git a/Modules/IO/TransformBase/wrapping/itkTransformFileReader.wrap b/Modules/IO/TransformBase/wrapping/itkTransformFileReader.wrap
index 9b52fbc..662df9e 100644
--- a/Modules/IO/TransformBase/wrapping/itkTransformFileReader.wrap
+++ b/Modules/IO/TransformBase/wrapping/itkTransformFileReader.wrap
@@ -1,3 +1,5 @@
+set(WRAPPER_AUTO_INCLUDE_HEADERS OFF)
+itk_wrap_include("itkTransformFileReader.h")
 itk_wrap_class("itk::TransformFileReaderTemplate" POINTER)
   foreach(t ${WRAP_ITK_REAL})
     itk_wrap_template("${ITKM_${t}}" "${ITKT_${t}}")
diff --git a/Modules/IO/TransformBase/wrapping/itkTransformFileWriter.wrap b/Modules/IO/TransformBase/wrapping/itkTransformFileWriter.wrap
index 4f1ac4a..356aaf3 100644
--- a/Modules/IO/TransformBase/wrapping/itkTransformFileWriter.wrap
+++ b/Modules/IO/TransformBase/wrapping/itkTransformFileWriter.wrap
@@ -1,3 +1,5 @@
+set(WRAPPER_AUTO_INCLUDE_HEADERS OFF)
+itk_wrap_include("itkTransformFileWriter.h")
 itk_wrap_class("itk::TransformFileWriterTemplate" POINTER)
   foreach(t ${WRAP_ITK_REAL})
     itk_wrap_template("${ITKM_${t}}" "${ITKT_${t}}")
-- 
1.8.5.3

